Makefile

```diff
@@ -0,0 +1,61 @@
+VERSION := 0.19.1
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# ABI versioning
+SONAME_MAJOR := 0
+SONAME_MINOR := 0
+
+OBJ := build/Release/obj.target/tree_sitter_go_binding/src/*.o
+
+# OS-specific bits
+ifeq ($(shell uname),Darwin)
+	SOEXT = dylib
+	SOEXTVER_MAJOR = $(SONAME_MAJOR).dylib
+	SOEXTVER = $(SONAME_MAJOR).$(SONAME_MINOR).dylib
+	LINKSHARED += -dynamiclib -Wl,-install_name,$(LIBDIR)/libtree-sitter-go.$(SONAME_MAJOR).dylib
+else
+	SOEXT = so
+	SOEXTVER_MAJOR = so.$(SONAME_MAJOR)
+	SOEXTVER = so.$(SONAME_MAJOR).$(SONAME_MINOR)
+	LINKSHARED += -shared -Wl,-soname,libtree-sitter-go.so.$(SONAME_MAJOR)
+endif
+ifneq (,$(filter $(shell uname),FreeBSD NetBSD DragonFly))
+	PCLIBDIR := $(PREFIX)/libdata/pkgconfig
+endif
+
+%.o:
+	npm install
```

Comment:
I don't think we should rely on NPM to build the parser here. Instead, I think we should just perform the compilation using Make's default rules (which run the default C compiler for `.c` files, and the default c++ compiler for `.cc` files).

By default, Make will put the `.o` files right next to the source files, which I think is fine. So, wherever you're currently referring to NPM's `build/Release` folder, you can instead just say `src/*.o`.

Reply:
Ohh, that would be way easier easier! I'm embarrassed to say I didn't even realize that was possible. I just followed the docs. And, I definitely struggled here, particularly when I needed to modify the gyp file.

Reply:
Changed to match what you suggested, and it's a million times better.